lib/src/obsoleted/rules/avoid_returning_widgets_rule.dart

```diff
@@ -0,0 +1,121 @@
+import 'package:analyzer/dart/analysis/results.dart';
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/dart/element/type.dart';
+import 'package:collection/collection.dart';
+
+import '../../models/issue.dart';
+import '../../models/severity.dart';
+import '../../utils/node_utils.dart';
+import '../../utils/rule_utils.dart';
+import 'obsolete_rule.dart';
+
+class AvoidReturningWidgets extends ObsoleteRule {
+  static const String ruleId = 'avoid-returning-widgets';
+  static const _documentationUrl = '';
+
+  static const _warningMessage = 'Avoid returning widgets from a function';
```

Comment:
@incendial I think a good idea use the rules of [parallel construction](https://en.wikipedia.org/wiki/Parallelism_(grammar))